src/shelly_hap_rgb.hpp

```diff
@@ -75,7 +89,12 @@ class RGBWLight : public Component, public mgos::hap::Service {
   std::vector<mgos::hap::Characteristic *> state_notify_chars_;
 
   mgos::Timer auto_off_timer_;
-  bool dirty_ = false;
+
+  mgos::Timer transition_timer_;
+  int64_t transition_start_;
```

Comment:
scalar class members should have defaults to ensure they are always initialized.

Reply:
> scalar class members should have defaults to ensure they are always initialized.

you are right, I forgot that

Reply:
I removed the HSV struct because I did not see any advantages with it anymore. The HSV base is now the cfg_ struct again as before.